test/parallel/test-fs-read-stream-throw-type-error.js

```diff
@@ -10,18 +14,23 @@ fs.createReadStream(example, null);
 fs.createReadStream(example, 'utf8');
 fs.createReadStream(example, { encoding: 'utf8' });
 
-const createReadStreamErr = (path, opt) => {
+const createReadStreamErr = (path, opt, errorCode) => {
   common.expectsError(
     () => {
       fs.createReadStream(path, opt);
     },
     {
-      code: 'ERR_INVALID_ARG_TYPE',
+      code: errorCode,
       type: TypeError
     });
 };
 
-createReadStreamErr(example, 123);
-createReadStreamErr(example, 0);
-createReadStreamErr(example, true);
-createReadStreamErr(example, false);
+createReadStreamErr(example, 123, 'ERR_INVALID_ARG_TYPE');
+createReadStreamErr(example, 0, 'ERR_INVALID_ARG_TYPE');
+createReadStreamErr(example, true, 'ERR_INVALID_ARG_TYPE');
+createReadStreamErr(example, false, 'ERR_INVALID_ARG_TYPE');
+
+// Should also throw on NaN (for https://github.com/nodejs/node/pull/19732)
+createReadStreamErr(example, { start: NaN }, 'ERR_OUT_OF_RANGE');
+createReadStreamErr(example, { end: NaN }, 'ERR_OUT_OF_RANGE');
+createReadStreamErr(example, { start: NaN, end: NaN }, 'ERR_OUT_OF_RANGE');
```

Comment:
Can you please still add tests for Infinity? :-) That is independent of making a breaking change. Setting start to Infinity will still throw right now and setting end to infinity should not throw.